app/components/scheduled_post_indicator/scheduled_post_indicator.test.tsx

```diff
@@ -0,0 +1,129 @@
+
+    it('should render single scheduled post indicator correctly', async () => {
+        const {getByText} = renderWithEverything(
+            <ScheduledPostIndicator
+                isThread={false}
+                scheduledPostCount={1}
+            />,
+            {database},
+        );
+
+        await screen.findByTestId('scheduled_post_indicator_single_time');
+        expect(getByText(/Message scheduled for/)).toBeTruthy();
+        expect(getByText(/See all./)).toBeTruthy();
+    });
+
+    it('should render multiple scheduled posts indicator for channel', async () => {
+        const {getByText} = renderWithEverything(
+            <ScheduledPostIndicator
+                isThread={false}
+                scheduledPostCount={10}
+            />,
+            {database},
+        );
+
+        expect(getByText(/10 scheduled messages in channel./)).toBeTruthy();
+        expect(getByText(/See all./)).toBeTruthy();
+    });
+
+    it('should render multiple scheduled posts indicator for thread', async () => {
+        const {getByText} = renderWithEverything(
+            <ScheduledPostIndicator
+                isThread={true}
+                scheduledPostCount={10}
+            />,
+            {database},
+        );
+
+        expect(getByText(/10 scheduled messages in thread./)).toBeTruthy();
+        expect(getByText(/See all./)).toBeTruthy();
+    });
+
+    it('renders with military time when preference is set', async () => {
+        await operator.handleConfigs({
+            configs: [
+                {id: 'use_military_time', value: 'true'},
+            ],
+            configsToDelete: [],
+            prepareRecordsOnly: false,
+        });
+
+        renderWithEverything(
+            <ScheduledPostIndicator
+                scheduledPostCount={1}
+            />,
+            {database},
+        );
+
+        const timeElement = await screen.findByTestId('scheduled_post_indicator_single_time');
+        expect(timeElement).toBeTruthy();
+    });
+
+    it('renders with 12-hour time when preference is not set', async () => {
+        await operator.handleConfigs({
+            configs: [
+                {id: 'use_military_time', value: 'false'},
+            ],
+            configsToDelete: [],
+            prepareRecordsOnly: false,
+        });
+
+        renderWithEverything(
+            <ScheduledPostIndicator
+                scheduledPostCount={1}
+            />,
+            {database},
+        );
+
+        const timeElement = await screen.findByTestId('scheduled_post_indicator_single_time');
+        expect(timeElement).toBeTruthy();
+    });
+
+    it('handles missing current user', async () => {
+        renderWithEverything(
+            <ScheduledPostIndicator
+                scheduledPostCount={1}
+            />,
+            {database},
+        );
+
+        const timeElement = await screen.findByTestId('scheduled_post_indicator_single_time');
+        expect(timeElement).toBeTruthy();
```

Comment:
Same, but how would I know the user is missing?

Reply:
Its just that it doesn't break when user is missing, no special behavior occurs on user missing that can be tested. Its a "just for safety" test, not useful.